### ISSUE TITLE
Symbolic representation for unbind 

### ISSUE THREAD

**Issue by iiSeymour** [pytorch/pytorch#22870]
I am trying to export a model to ONNX that uses [unbind](https://pytorch.org/docs/stable/torch.html#torch.unbind) which is not a supported op.
> RuntimeError: ONNX export failed: Couldn't export operator aten::unbind

What would be the symbolic definition for unbind?

**Comment by zou3519** [comment]
cc @houseroad @bddppq

**Comment by iiSeymour** [comment]
Unbind results in `aten::ATen[operator="unbind"]` and a `prim::ListUnpack`.

```python
import onnxruntime
from io import BytesIO
from torch.onnx import OperatorExportTypes

class Model(torch.nn.Module):
    def forward(self, x):
        return x.unbind(0)

x = torch.tensor([[1., 2., 3.], [4., 5., 6.], [7., 8., 9.]])

torch.onnx.export(
    Model(), x, BytesIO(), verbose=True,
    operator_export_type=OperatorExportTypes.ONNX_ATEN_FALLBACK
)
```

I know how to register symbolic handlers for `aten` ops but I'm not sure what to do with `prim`. 

```python
graph(%0 : Float(3, 3)):
  %1 : Long() = onnx::Constant[value={0}](), scope: Model
  %2 : Tensor[] = aten::ATen[operator="unbind"](%0, %1), scope: Model
  %3 : Float(3), %4 : Float(3), %5 : Float(3) = prim::ListUnpack(%2), scope: Model
  return (%3, %4, %5)
```

**Comment by houseroad** [comment]
There is no tensor list concept in ONNX. With this concept, it's pretty hard to represent the results of unbind in onnx, unless your output number of unbind op can be determined at compile time.

cc: @spandantiwari @BowenBao

**Comment by iiSeymour** [comment]
For the particular model i'm looking at; the operation doesn't have an effect on the result so I could just monkey patch it away in my conversion script.

```python
torch.onnx.symbolic.unbind = lambda g, x, d: x
torch.onnx.symbolic.listunpack = lambda g, x, d: x 
```

This works for unbind but leaves the list unpack.

```python
graph(%0 : Tensor[]):
  %1 : Float(3), %2 : Float(3), %3 : Float(3) = prim::ListUnpack(%0), scope: Model
  return (%1, %2, %3)
```

**Comment by BowenBao** [comment]
@iiSeymour like Lu suggested, if the number of outputs is known, one way to work around this is using split.

```python
class Model(torch.nn.Module):
    def forward(self, x):
        return [torch.squeeze(out, 0) for out in torch.split(x, [1,1,1], dim=0)]#x.unbind(0)

x = torch.tensor([[1., 2., 3.], [4., 5., 6.], [7., 8., 9.]])
```

**Comment by iiSeymour** [comment]
@BowenBao it’s a pretrained model I’m trying to convert.

**Comment by spandantiwari** [comment]
@iiSeymour - Full support for exporting torch.unbind to ONNX needs a list style datatype support in ONNX. A PR for that is under review and after that we need to get some list based operator support in ONNX. That is ongoing work. Once that is in place in ONNX, we will add support for torch.unbind.

**Comment by iiSeymour** [comment]
Thanks @spandantiwari. The uses of `unbind` in my model of interest seem to be for performance reasons only, so the whole tensor doesn’t get evaluated unnecessarily in the call to `backwards`. A method to prune the `prim::ListUnpack` from the graph would help.

**Comment by garymm** [comment]
This seems to work on latest nightly PyTorch. If there's still a problem please add details here or open a new issue.

```python
import torch
print("torch version:", torch.__version__)

class Model(torch.nn.Module):
    def forward(self, x):
        return x.unbind(0)

x = torch.tensor([[1., 2., 3.], [4., 5., 6.], [7., 8., 9.]])
torch.onnx.export(Model(), x, "/dev/null", verbose=True)
```

Outputs:
```
torch version: 1.11.0.dev20211221+cpu
graph(%0 : Float(3, 3, strides=[3, 1], requires_grad=0, device=cpu)):
  %1 : Float(1, 3, strides=[3, 1], device=cpu), %2 : Float(1, 3, strides=[3, 1], device=cpu), %3 : Float(1, 3, strides=[3, 1], device=cpu) = onnx::Split[axis=0, split=[1, 1, 1]](%0) # <ipython-input-4-bc95856ba5dc>:7:0
  %4 : Float(3, strides=[1], requires_grad=0, device=cpu) = onnx::Squeeze[axes=[0]](%1) # <ipython-input-4-bc95856ba5dc>:7:0
  %5 : Float(3, strides=[1], requires_grad=0, device=cpu) = onnx::Squeeze[axes=[0]](%2) # <ipython-input-4-bc95856ba5dc>:7:0
  %6 : Float(3, strides=[1], requires_grad=0, device=cpu) = onnx::Squeeze[axes=[0]](%3) # <ipython-input-4-bc95856ba5dc>:7:0
  return (%4, %5, %6)
```
